src/CppAst/CppModelBuilder.cs

```diff
@@ -1212,43 +1245,47 @@ private static CppStorageQualifier GetStorageQualifier(CXCursor cursor)
             return CppStorageQualifier.None;
         }
 
-        private CppFunction VisitFunctionDecl(CXCursor cursor, CXCursor parent, void* data)
+private CppFunction VisitFunctionDecl(CXCursor destinationCursor, CXCursor cursor, CXCursor parent, void* data)
         {
+            var destinationContextContainer = GetOrCreateDeclarationContainer(destinationCursor.SemanticParent, data);
+            var destinationContainer = destinationContextContainer.DeclarationContainer;
             var contextContainer = GetOrCreateDeclarationContainer(cursor.SemanticParent, data);
             var container = contextContainer.DeclarationContainer;
 
-            if (container == null)
+            if (destinationContainer == null || container == null)
             {
                 WarningUnhandled(cursor, parent);
                 return null;
             }
 
-            var functionName = GetCursorSpelling(cursor);
+            // This (not using 'cursor', but 'destinationCursor') is only used when we pulling in constructors from Foo, into Bar
+            // We are renming effectively the pulled in 'Foo' constructor into 'Bar'
```

Comment:
renming -> renaming? (If yes, what do you mean by renaming the ctor into another class?)

Reply:
Yeah, exactly, with "using" if you have Bar : public Foo, when "using" is parsed in "Bar", you'll get Foo's constructors, e.g. "Foo::Foo", so when we add those to "Bar", we need to rename them to "Bar::Bar"

Reply:
Aok:)